homeassistant/components/ccm15/coordinator.py

```diff
@@ -0,0 +1,247 @@
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=host,
+            update_method=self._async_update_data,
+            update_interval=datetime.timedelta(seconds=interval),
+        )
+        self._ccm15 = CCM15Device(host, port, DEFAULT_TIMEOUT)
+        self._host = host
+        self._ac_devices: dict[int, CCM15Climate] = {}
+
+    def get_devices(self):
+        """Get all climate devices from the coordinator."""
+        return self._ac_devices.values()
+
+    async def _async_update_data(self) -> CCM15DeviceState:
+        """Fetch data from Rain Bird device."""
+        try:
+            return await self._fetch_data()
+        except httpx.RequestError as err:  # pragma: no cover
+            raise UpdateFailed(f"Error communicating with Device: {err}") from err
+
+    async def _fetch_data(self) -> CCM15DeviceState:
+        """Get the current status of all AC devices."""
+        ac_data = await self._ccm15.get_status_async()
+        if len(self._ac_devices) == 0:
+            for ac_index in ac_data.devices:
+                _LOGGER.debug("Creating new ac device at index '%s'", ac_index)
+                self._ac_devices[ac_index] = CCM15Climate(self._host, ac_index, self)
+        return ac_data
+
+    async def async_test_connection(self):  # pragma: no cover
+        """Test the connection to the CCM15 device."""
+        return await self._ccm15.async_test_connection()
+
+    async def async_set_state(self, ac_index: int, state: str, value: int) -> None:
+        """Set new target states."""
+        if await self._ccm15.async_set_state(ac_index, state, value):
+            await self.async_request_refresh()
+
+    def get_ac_data(self, ac_index: int) -> CCM15SlaveDevice | None:
+        """Get ac data from the ac_index."""
+        if ac_index < 0 or ac_index >= len(self.data.devices):
+            # Index is out of bounds or not an integer
+            _LOGGER.warning(
+                "Data is not available for AC id %s.%s", self._host, ac_index
+            )
+            return None
+        data = self.data.devices[ac_index]
+        return data
+
+    async def async_set_hvac_mode(self, ac_index, hvac_mode: HVACMode) -> None:
+        """Set the hvac mode."""
+        _LOGGER.debug("Set Hvac[%s]='%s'", ac_index, str(hvac_mode))
+        await self.async_set_state(ac_index, "mode", CONST_STATE_CMD_MAP[hvac_mode])
+
+    async def async_set_fan_mode(self, ac_index, fan_mode: str) -> None:
+        """Set the fan mode."""
+        _LOGGER.debug("Set Fan[%s]='%s'", ac_index, fan_mode)
+        await self.async_set_state(ac_index, "fan", CONST_FAN_CMD_MAP[fan_mode])
+
+    async def async_set_temperature(self, ac_index, temp) -> None:
+        """Set the target temperature mode."""
+        _LOGGER.debug("Set Temp[%s]='%s'", ac_index, temp)
+        await self.async_set_state(ac_index, "temp", temp)
+
+
+class CCM15Climate(CoordinatorEntity[CCM15Coordinator], ClimateEntity):
```

Comment:
I cannot place CCM15 into its own file, it will fail to load with:
```
ImportError: cannot import name 'CCM15Coordinator' from partially initialized module 'homeassistant.components.ccm15.coordinator' (most likely due to a circular import) (/workspaces/home-assistant/homeassistant/components/ccm15/coordinator.py)
```